hash_practice/exercises.py

```diff
@@ -2,18 +2,49 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagrams = {}
+    for word in strings:
+        alphabet = [0] * 26
+        for letter in word:
+            alphabet[ord(letter.lower()) - ord("a")] += 1
+        if tuple(alphabet) in anagrams:
+            anagrams[tuple(alphabet)].append(word)
+        else:
+            anagrams[tuple(alphabet)] = [word]
+
+    return_list = []
+    for k, v in anagrams.items():
+        return_list.append(v)
+    return return_list
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
Note that `max` may end up going through every key-value pair in `element_dict`, of which there may be O(n) pairs, one for every number in `nums`. It may do this up to `k` times so the time complexity here is O(n + n*k), of which that last part dominates for O(n*k).